### ISSUE TITLE
Cannot install package

### ISSUE THREAD

**Issue by rawriddims** [CloudCannon/pagefind#66]
I'm building in my package.json like this:

```
  "scripts": {
    "all": "npm run build && npm run postbuild",
    "build": "npx @11ty/eleventy",
    "postbuild": "npx pagefind --source _site"
  },
```
and then `npm run all --serve`

I'm doing this locally, in dev mode, at the moment, and here's the output:

```
> rawriddims@1.0.0 all
> npm run build && npm run postbuild


> rawriddims@1.0.0 build
> npx @11ty/eleventy

Writing _site/... etc etc
Copied 841 files / Wrote 72 files in 25.28 seconds (351.1ms each, v0.12.1)

> rawriddims@1.0.0 postbuild
> npx pagefind --source _site

Need to install the following packages:
  pagefind
Ok to proceed? (y) y

> rawriddims@1.0.0 postbuild
> npx pagefind --source _site
```
And it stops like this. I do not see any Pagefind output of indexed and created, and no /_pagefind/ directory is in my _site.

I do have the Pagefind search UI in my search page:
```
          <div id="search"></div>
          <script>
              window.addEventListener('DOMContentLoaded', (event) => {
                  new PagefindUI({ element: "#search",
                  showImages: false,
                  });
              });
          </script>
```
and I just omitted `<link href="/_pagefind/pagefind-ui.css" rel="stylesheet">`;
while  `<script src="/_pagefind/pagefind-ui.js"></script>`  is in my 11ty _includes layout with other JS that work fine.

I've also tried to run separately in my Windows Power Shell `npx -y pagefind --source _site --serve` but no success: 

What am I missing? Thanks!

**Comment by TimGeisler** [comment]
I have the same problem with Windows. Try to run directly the executable `node_modules/pagefind/bin/pagefind_extended`.

**Comment by rawriddims** [comment]
You mean `npx run node_modules/pagefind/bin/pagefind_extended` and then run it in my package.json like the above?

**Comment by TimGeisler** [comment]
No, first install pagefind by adding it to your devDependencies. Then try to execute the executable in this folder directly (e.g. in your Powershell). That works for me.
However, we still need to find out the exact reason of the problems for Windows. There are more than one way where you can run the npx script: in cmd, in git bash, in cygwin, in PowerShell. I guess some parts of the extraction/recognition logic do not handle all those different settings. I need to find some time to look into that the next days.

**Comment by rawriddims** [comment]
So, I did `npm install pagefind --save-dev` and here it is in the package,json:
```
  "devDependencies": {
    "@11ty/eleventy": "^0.12.1",
    "pagefind": "^0.8.0"
  },
```
Then I had to navigate down upto `\node_modules\pagefind\bin` and run the binary directly `./pagefind_extended --source "_site"`
But nothing happens, except a stupid Windows prompt which asks me "How do you like to open this file?".

Obviously I'm doing something wrong,

**Comment by TimGeisler** [comment]
I tried to install pagefind on Windows 10 with 4 different shells: cmd, Powershell, git bash, cygwin bash.
For each of the cases, I created a new NPM package using `npm init` and accepting all defaults.
With cmd and Powershell, installation worked fine (node_modules was created, pagefind folder exists).
On both git bash and cygwin bash, I got similar error messages:
```
$ npm install --save-dev pagefind
npm ERR! code 1
npm ERR! path C:\Users\tg\pagefind\gitbash\node_modules\pagefind
npm ERR! command failed
npm ERR! command C:\WINDOWS\system32\cmd.exe /d /s /c node ./lib/postinstall.js
npm ERR! Finding release for v0.8.0
npm ERR! https://api.github.com/repos/CloudCannon/pagefind/releases/tags/v0.8.0
npm ERR! GET https://api.github.com/repos/CloudCannon/pagefind/releases/tags/v0.8.0
npm ERR! Downloading from https://api.github.com/repos/CloudCannon/pagefind/releases/assets/75547266
npm ERR! Downloading to C:\Users\tg\pagefind\gitbash\node_modules\pagefind\cache\0.8.0\pagefind_extended-v0.8.0-x86_64-pc-windows-msvc.tar.gz
npm ERR! Downloading with Invoke-WebRequest
npm ERR! Unzipping to C:\Users\tg\pagefind\gitbash\node_modules\pagefind\bin
npm ERR! tar xvf exited with 128
npm ERR! Deleting invalid download
npm ERR! tar: Cannot connect to C: resolve failed
npm ERR! Downloading pagefind failed: Error: tar xvf exited with 128
npm ERR!     at ChildProcess.<anonymous> (C:\Users\tg\pagefind\gitbash\node_modules\pagefind\lib\download.js:217:24)
npm ERR!     at ChildProcess.emit (node:events:390:28)
npm ERR!     at maybeClose (node:internal/child_process:1064:16)
npm ERR!     at Process.ChildProcess._handle.onexit (node:internal/child_process:301:5)
```
Thus, it seems that the unpacking of the tar.gz file does not work on Windows with git bash/cygwin bash.

**Comment by TimGeisler** [comment]
When I try to run the created file `node_modules/pagefind/bin/pagefind_extended` in both cmd and Powershell, I get the behaviour which @rawriddims observed.

However, when I try to run these files with bash (git or cygwin), it works:
![grafik](https://user-images.githubusercontent.com/11179787/188493512-71ed5bd7-9008-4cde-bf9b-20ee219c145e.png)

**Comment by TimGeisler** [comment]
Unfortunately, I am not an expert to get those install/postinstall/... scripts working on all different shells on Windows. The combination of installing/unpacking with cmd/Powershell and running with bash worked for me. This helps for trying it out, but is not a solution if you develop on Windows.

**Comment by TimGeisler** [comment]
When you rename `pagefind_extended` to `pagefind_extended.exe`, it works with cmd and Powershell, respectively:
![grafik](https://user-images.githubusercontent.com/11179787/188494131-269b8c10-bb55-4f33-804d-5e4a0f885645.png)

**Comment by TimGeisler** [comment]
The reason why the `.exe` extension is missing is that the tar files for Windows contain the executables without `.exe` extension.

**Comment by rawriddims** [comment]
It's a bit too cumbersome for me, as I have to develop on Windows. Maybe I should wait for the .exe extension to be added to the Windows executables, or maybe for a plain Windows installation to be added -  if that's in the plans for the future. I look forward to see such developments of this excellent library. Thank you.

**Comment by bglw** [comment]
Thanks for the deep dive @TimGeisler 🙏 I'm away from my keyboard this week but I will look at tidying that up when I'm back in a few days 🙂 

It looks like changing the files in the tar archive to end in `.exe` will fix one problem. Additionally the `tar xvf` throwing an error needs to be looked into — I might swap that command out for a library like [tar](https://www.npmjs.com/package/tar) to try and sidestep any platform/shell specific errors.

**Comment by bglw** [comment]
Sorry for the delay all — this should be fixed by https://github.com/CloudCannon/pagefind/pull/113 — aiming for a release this week 🙂

**Comment by bglw** [comment]
👋 Hello! This is fixed in the [Pagefind v0.9.1](https://github.com/CloudCannon/pagefind/releases/tag/v0.9.1) release 🎉

**Comment by whyboris** [comment]
Thank you @TimGeisler 🙇  for solving the error I was encountering: `Error: tar xvf exited with 128`

With your comment above: https://github.com/CloudCannon/pagefind/issues/66#issuecomment-1237313541

Turns out I needed to run the `npm install` command with **Powershell** rather than _GitBash_ 😅 

@bglw -- would you consider adding a note[^1] in the _Installing and running Pagefind_ section of your website: https://pagefind.app/docs/installation/

In fact, I'm surprised you don't also share (for the new developers not familiar with all the `npm` and `node` stuff) that you can run `npm install` and not just `npx` 🤔 

[^1]: Something like this: If you see an[Error: tar xvf exited with 128](https://github.com/CloudCannon/pagefind/issues/66#issuecomment-1237313541) during installation, try using _Powershell_ ❤️
